From 2a093769a29f03103195b34c269411ee21b646e2 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Wed, 20 Jan 2021 10:37:09 +0800
Subject: [PATCH 21/49] LF-3272: drm: cdns_mhdp: fix Coverity Issue: 11566406

Add default access hdcp bus to fix
Coverity Issue: 11566406 Uninitialized scalar variable.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
---
 drivers/gpu/drm/bridge/cadence/cdns-mhdp-hdcp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/bridge/cadence/cdns-mhdp-hdcp.c b/drivers/gpu/drm/bridge/cadence/cdns-mhdp-hdcp.c
index 587c5f953489..b3c931382013 100644
--- a/drivers/gpu/drm/bridge/cadence/cdns-mhdp-hdcp.c
+++ b/drivers/gpu/drm/bridge/cadence/cdns-mhdp-hdcp.c
@@ -27,6 +27,8 @@ static u32 mhdp_hdcp_bus_read(struct cdns_mhdp_device *mhdp, u32 offset)
 		val = readl((offset & 0xfff) + mhdp->regs_base);
 	} else if (mhdp->bus_type == BUS_TYPE_NORMAL_APB)
 		val = readl(mhdp->regs_sec + offset);
+	else
+		val = readl(mhdp->regs_base + offset);
 
 	mutex_unlock(&mhdp->iolock);
 
-- 
2.29.2

